x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/es_geo_grid_source.tsx

```diff
@@ -46,45 +60,56 @@ export const heatmapTitle = i18n.translate('xpack.maps.source.esGridHeatmapTitle
   defaultMessage: 'Heat map',
 });
 
-export class ESGeoGridSource extends AbstractESAggSource {
-  static type = SOURCE_TYPES.ES_GEO_GRID;
-
-  static createDescriptor({ indexPatternId, geoField, metrics, requestType, resolution }) {
+export class ESGeoGridSource extends AbstractESAggSource implements ITiledSingleLayerVectorSource {
+  static createDescriptor(
+    descriptor: Partial<ESGeoGridSourceDescriptor>
+  ): ESGeoGridSourceDescriptor {
+    descriptor = AbstractESAggSource.createDescriptor(descriptor);
     return {
-      type: ESGeoGridSource.type,
+      ...descriptor,
+      type: SOURCE_TYPES.ES_GEO_GRID,
       id: uuid(),
-      indexPatternId,
-      geoField,
-      metrics: metrics ? metrics : [DEFAULT_METRIC],
-      requestType,
-      resolution: resolution ? resolution : GRID_RESOLUTION.COARSE,
+      indexPatternId: descriptor.indexPatternId || '',
+      geoField: descriptor.geoField || '',
+      metrics:
+        descriptor.metrics && descriptor.metrics.length > 0 ? descriptor.metrics : [DEFAULT_METRIC],
+      requestType: descriptor.requestType || RENDER_AS.POINT,
+      resolution: descriptor.resolution ? descriptor.resolution : GRID_RESOLUTION.COARSE,
     };
   }
 
-  constructor(descriptor, inspectorAdapters) {
-    super(descriptor, inspectorAdapters, descriptor.resolution !== GRID_RESOLUTION.SUPER_FINE);
+  readonly _descriptor: ESGeoGridSourceDescriptor;
+
+  constructor(descriptor: Partial<ESGeoGridSourceDescriptor>, inspectorAdapters?: Adapters) {
+    const sourceDescriptor = ESGeoGridSource.createDescriptor(descriptor);
+    super(
+      sourceDescriptor,
+      inspectorAdapters,
+      descriptor.resolution !== GRID_RESOLUTION.SUPER_FINE
+    );
+    this._descriptor = sourceDescriptor;
   }
 
-  renderSourceSettingsEditor({ onChange, currentLayerType }) {
+  renderSourceSettingsEditor(sourceEditorArgs: SourceEditorArgs): ReactElement<any> {
     return (
       <UpdateSourceEditor
-        currentLayerType={currentLayerType}
+        currentLayerType={sourceEditorArgs.currentLayerType}
         indexPatternId={this.getIndexPatternId()}
-        onChange={onChange}
+        onChange={sourceEditorArgs.onChange}
         metrics={this._descriptor.metrics}
         renderAs={this._descriptor.requestType}
         resolution={this._descriptor.resolution}
       />
     );
   }
 
-  getSyncMeta() {
+  getSyncMeta(): VectorSourceSyncMeta {
     return {
-      requestType: this._descriptor.requestType,
+      requestType: this._descriptor.requestType || RENDER_AS.POINT,
```

Comment:
requestType will always be provided at this point so just use not null operator or remove `?` in `ESGeoGridSourceDescriptor`